docs/source/DeveloperGuide/APIs/index.rst

```diff
@@ -0,0 +1,111 @@
+
+What is the purpose of this?
+----------------------------
+
+The purpose of this component is to build DIRAC system status access interfaces based on HTTP requests outside the fixed RPC request interface described in TornadoServices.
+
+The main reason for the implementation of this feature is the need to implementation of API for DIRAC Authorization Server based on OAuth2 framework,
+which in turn is an integral part of the implementation of OAuth2 DIRAC authorization.
+So currently, there is only one API implementation for the Framework system, which is :py:class:`~ DIRAC.FrameworkSystem.API.AuthHandler`.
+For the same reason, the ability to authorize with an access token (see https://datatracker.ietf.org/doc/html/rfc6750#section-2) was added to the authorization steps.
+
+How to write APIs
+-----------------
+
+The following is an example of a simple API:
+
+.. literalinclude::
+
+    from DIRAC.Core.Tornado.Server.TornadoREST import TornadoREST
+    from DIRAC.ConfigurationSystem.Client.Helpers import Registry
+    class yourEndpointHandler(TornadoREST):
+
+      SYSTEM = "Configuration"
+      LOCATION = "/registry"
+
+      @classmethod
+      def initializeHandler(cls, infosDict):
+        ''' Called only once when the first request for this handler arrives
+        '''
+        pass
+
+      def initializeRequest(self):
+        ''' Called at the beginning of each request
+        '''
+        pass
+
+      # Specify the default permission for the method
+      auth_users = ['authenticated']
+
+      def web_users(self, count: int = 0):
+        ''' Your method. It will be available for queries such as https://domain/registry/users?count=1
+
+            TornadoREST will try to convert the received argument `count` to int.
+
+        '''
+        return Registry.getAllUsers()[:count]
+
+.. note:: For the better look of your requests you can listen default https port (443), balancer like a Nginx can help you with this.
+
+The example described is likely to be sufficient for most writing cases. But here are some additional features, see :py:class:`~ DIRAC.Core.Tornado.Server.private.BaseRequestHandler`:
+
+  - ``USE_AUTHZ_GRANTS`` set the list and order of steps to authorize the request. For example, set ``USE_AUTHZ_GRANTS = ["JWT"]`` to allow access to your endpoint only with a valid access token.
+  - ``AUTH_PROPS`` set the authorization requirements. For example, ``AUTH_PROPS = ['authenticated']`` will allow access only to authenticated users.
+  - in addition to standard S_OK/S_ERROR you can return text, whether the dictionary for example or nothing, the result will be sent with a 200 status.
+  - ``path_<my_method>`` will allow you to consider the path parameters as positional arguments of the target method. For example:
+
+    .. code-block:: python
+
+        # It's allow make request like a GET /user/contacts/Bob
+        path_user = ['[A-z]+', '[A-z]+']
```

Comment:
do you mean 
```suggestion
        path_user = ["([A-z%0-9-_]*)"]
```
?

Reply:
yes, thanks. I fixed it to correct regex.